relay-metrics/src/aggregation.rs

```diff
@@ -1782,22 +1785,35 @@ impl Aggregator {
             let partitioned_buckets = self.partition_buckets(project_buckets, num_partitions);
             for (partition_key, buckets) in partitioned_buckets {
                 self.process_batches(buckets, |batch| {
-                    let batch_size = batch.len();
-                    match self.receiver.do_send(FlushBuckets {
+                    if let Err(err) = self.receiver.do_send(FlushBuckets {
                         project_key,
                         partition_key,
                         buckets: batch,
                     }) {
-                        Ok(_) => {}
-                        Err(err) => {
-                            relay_log::error!("Failed to flush the buckets: {err}. Dropping current bucket batch of size: {batch_size}");
-                        }
+                        let FlushBuckets {
+                            project_key,
+                            buckets,
+                            ..
+                        } = err.into_inner();
+
+                        // remove the number of buckets which were not flushed due to the `SendError`
+                        total_bucket_count -= buckets.len() as u64;
+                        buckets_failed.push((project_key, buckets));
```

Comment:
Why not call `merge_all` right away here?

Reply:
So basically we already use `self` in the closure, and if we want to call `merge_all` on self again, which again uses mutable reference, the compiler is not happy. 

But I will take another look if I can refactor this a little more.

Reply:
There are now other places where can fail to send buckets along, so is it even worth capturing them here?

Reply:
reverted this change to just logging of the error, and merged `master` to see if all the checks passing now on CI